src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1118,19 +1128,130 @@ map_injective f.injective u v
 
 end path
 
+/-! ## Transferring between graphs -/
+
+namespace walk
+
+variables {G}
+
+/-- The walk `p` transferred to lie in `H` -/
+@[protected, simp] def transfer : Π {u v : V} (p : G.walk u v) (H : simple_graph V)
+  (h : ∀ e, e ∈ p.edges → e ∈ H.edge_set), H.walk u v
+| _ _ (walk.nil) H h := walk.nil
+| _ _ (walk.cons a p) H h := by
+  { refine walk.cons _ (p.transfer H _);
+    simp only [walk.edges_cons, list.mem_cons_iff, forall_eq_or_imp, mem_edge_set] at h,
+    exact h.1, exact h.2, }
```

Comment:
Here's a shorter way to write this case. I prefer it because it doesn't use tactic mode for the definition, but that's not a strict mathlib rule.
```suggestion
| _ _ (walk.cons' u v w a p) H h :=
  walk.cons (h ⟦(u, v)⟧ (by simp)) (p.transfer H (λ e he, h e (by simp [he])))
```